submissions/JuliaNovak/tiny-js-world/index.js

```diff
@@ -0,0 +1,70 @@
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitant {
+   constructor(species, name, gender, legs, hands, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.hands = hands;
+      this.saying = saying;
+   }
+
+   display() {
+      console.log(this);
+      let values = "";
+      for (const prop in this) {
+         values += ` ${this[prop]}`;
+      }
+      print(values);
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying) {
+      super("human", name, gender);
```

Comment:
What `super` is and what it does? How did you decide to use it this way?
This is not a rhetorical question.

Reply:
I've answered this question above: "Thanks for the links! The problem is that I've read these articles a couple of times but I still don't see the mistake. Maybe u could point to it or give some more hints? Answering the question, I think that super here acts as a function and it references the variables in the parent class"
I've read that there are two ways of using _super_: the one is represented in the code snippet above and another one is super.methodName(). I suppose that the first variant is used to reference the variables in the parent class and the second one - to reference the methods in the parent class. Since I need variables, I chose the 1st variant

Reply:
What happens when `super` is called in the child class' constructor? What is called actually?

Reply:
it calls the parent's constructor as I understand

Reply:
may I also say that it calls the parent's prototype? or is it not correct?